packages/eslint-plugin/src/rules/no-confusing-void-expression.ts

```diff
@@ -112,6 +139,20 @@ export default createRule<Options, MessageId>({
         if (invalidAncestor.type === AST_NODE_TYPES.ArrowFunctionExpression) {
           // handle arrow function shorthand
 
+          if (options.ignoreVoidInVoid) {
+            if (
+              (invalidAncestor.returnType &&
+                isVoidLikeType(invalidAncestor.returnType)) ||
+              isValidFunctionExpressionReturnType(invalidAncestor, {
+                allowDirectConstAssertionInArrowFunctions: true,
```

Comment:
```suggestion
```

[Refactor] Unit tests still pass when this line is removed. Do we really need it? As far as I know, TypeScript will complain if we do `console.log() as const`